### PR TITLE
Clarify `plugin_name` and `display_name` in the prompts docs and defaults

### LINKED ISSUE #126: Change the default plugin name from `napari-foobar` to anything that doesn't start with `napari`
Since we no longer need the `napari-` prefix to find plugins, and it's not a very useful prefix to have when browsing for plugins, we should change the default plugin name to something else, to encourage new users to *not* have `napari` as the first word in their package name.

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
This PR aims to close #126 in light of napari PRs:
https://github.com/napari/napari/pull/5305
https://github.com/napari/napari/pull/5351
which addressed https://github.com/napari/napari/issues/5304 in napari repo
as well as the fact that the napari-hub is using `display_name`:
https://github.com/chanzuckerberg/napari-hub/issues/759#issuecomment-1319221129

Now `display_name` really is used for user-facing text, in both the Plugins menu in napari and in the File > Open Sample menu, in this PR the docs are clarified to emphasize where `display_name` used. Additionally, the defaults are tweaked to encourage `display_name` use by making the `display_name` more descriptive and human readable.

### CODE REVIEW

File: PROMPTS.md
Comment (DragaDoncila) [inline PROMPTS.md:56]:
```suggestion
for user-facing text, napari and the napari hub will use the `display_name` (below).
```

File: PROMPTS.md
Comment (DragaDoncila) [inline PROMPTS.md:80]:
```suggestion
as by the napari hub. It should be 3-40 characters long. It will be listed in
```

**Review by DragaDoncila — APPROVED** [review]
Perfect, thank you! Super minor nit as napari hub is not hyphenated. Test failures look unrelated so merge at will!

**Review by nclack — APPROVED** [review]
lgtm! thanks!

**Comment by psobolewskiPhD** [comment]
This repo doesn't appear to have releases, so if I am understanding correctly, once this is merged it will be live for making plugins, but yet not be true for the case of napari. So maybe best to hold off until there is at least a timetable for the napari release that could provide the functionality advertised? (currently broken too, because i messed up with the PRs 👀 see: https://github.com/napari/napari/pull/5366?notification_referrer_id=NT_kwDOBJEpGbM0OTEwOTM2NzU2Ojc2NjIyMTA1&notifications_query=repo%3Anapari%2Fnapari#issuecomment-1326010742)

**Comment by DragaDoncila** [comment]
oooops I hadn't refreshed the page and missed that last comment @psobolewskiPhD 🤦 - I don't mind this though. I wanted to update that prompt for a while, and it's still true that this name will be GUI facing. I don't think we should worry about reverting this

**Pull request merged**
